logging/cluster-logging-external.adoc

```diff
@@ -24,6 +24,8 @@ Alternatively, you can create a ConfigMap to use the xref:../logging/cluster-log
 You cannot use the ConfigMap methods and the Log Forwarding API in the same cluster.
 ====
 
+If your external logging aggregator becomes unavailable and cannot receive logs, Fluentd continues to collect logs and stores the logs in a buffer. When the log aggregator becomes available, log forwarding resumes, including the buffered logs. The buffer can grow to 10% of the size of the disk in the Fluentd node host. If the buffer fills completely, Fluentd stops collecting logs. Fluentd rotates the logs and deletes them. You cannot adjust the buffer size or add a persistent volume claim (PVC) to the Fluentd daemon set or pods.
```

Comment:
You could change "Fluentd continues to collect logs and stores the logs in a buffer" to "Fluentd continues to collect logs and stores them in a buffer", to reduce the number of references to "logs" in that sentence.

Reply:
I was a bit unsure about how these two sentences relate to each other:

"If the buffer fills completely, Fluentd stops collecting logs. Fluentd rotates the logs and deletes them."

Does Fluentd stop collecting logs when the buffer is full, or does it continue to collect logs by overwriting existing buffer data? I think it would be useful to clarify what is meant here, for the reader.

Reply:
In this sentence, does "the disk" refer to all internal storage in the Fluentd or one disk in particular?:

"The buffer can grow to 10% of the size of the disk in the Fluentd node host."

It might be worth stating "overall storage capacity" or something like that instead of disk, for clarity.

Also, you could consider removing either "node" or "host" from the same sentence. For example :

"The buffer can grow to 10% of the size of the disk in the Fluentd node."

"The buffer can grow to 10% of the size of the disk in the Fluentd host."

Reply:
It is the disk on Openshift worker .  fluentd mount the /var/lib/fluentd, and write the bufferfile in it.

Reply:
@mburke5678 

If the external logging aggregator becomes unavailable and stop receiving logs. Fluentd continues to collect logs and stores the logs in the bufferfiles. The bufferfiles are on the disk in the Openshift nodes.  Once the log aggregator becomes available, log forwarding resumes.  The buffer can grow up to 15% of the disk for each output type.  if there are multiple output plugin, the disk occupation may be multiple. If the buffersfills completely, Fluentd rotates the buffer files. if the logs wasn't be sent out before rotated, you may lost them forever. you can change the buffer size using fluentd Environment in Unmanagement status.

Reply:
@jcantrill   "The buffer can grow up to 15% of the disk for each output type", is this right? Does the retry make the buffer size  larger?

Reply:
@anpingli  @jcantrill If the buffer fills and Fluentd rotates off logs, does that create space in the buffer for Fluentd to create more logs?

Reply:
@sichvoge I assume that we don't want to mention that the user can go unmanaged (unsupported) in order to chage the buffer size?

Reply:
@jcantrill Can you help? 
 * The buffer can grow to 10% of the size of the disk in the Fluentd node host.
This is the size of the hard drive in the server where the Fluentd node is scheduled to?

* When Fluentd loses contact with an external aggregator, it stores logs in a buffer. When the buffer is full (10 or 15%), Fluentd rotates and deletes logs in the buffer. Does that free up space in the buffer for Fluentd to again store logs there?

Reply:
Nope, when the buffer is full fluentd stops (and logs a "buffer full" error)
File rotation is done by the node, before fluentd reads the files.

IMO the buffers are much too big. We could uses much smaller fixed size buffers (need to experiment to find the best level) The log files on disk are already buffering the log data for us, so there's no value in buffering again inside fluentd. Our large buffers are only ensuring that we save old messages while new ones are rotated away and lost. But this is probably not the right PR for that :)

Reply:
@alanconway Do you know if the buffers max out at 10 or 15%?